src/grt/src/GlobalRouter.cpp

```diff
@@ -661,10 +661,16 @@ void GlobalRouter::findPins(Net* net, std::vector<RoutePt>& pinsOnGrid)
       }
 
       if (!invalid) {
-        pinsOnGrid.push_back(RoutePt(pinX, pinY, topLayer));
+        RoutePt onGrid = RoutePt(pinX, pinY, topLayer);
+        pinsOnGrid.push_back(onGrid);
+        if (pin.isDriver() && net->getSignalType() == odb::dbSigType::CLOCK) {
+          root_idx = pinsOnGrid.size()-1;
+        }
       }
     }
   }
+
+  return root_idx;
```

Comment:
why does findPins return the root index?
if you are trying to return 2 values (ie, pinsOnGrid that should be pins_on_grid), then pass it as a reference (my preference) or a pointer (google preference) in the arg list